Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -164,7 +197,14 @@ void EventRegionContext::uniteInteractionRegions(const Region& region, RenderObj
                 return;
             m_occlusionRects.add(bounds);
         }
-
+        auto occlusionRect = guardRectForRegionBounds(interactionRegion->rectInLayerCoordinates);
```

Comment:
we could have a `Type::Occlusion` here and don't want to generate a guard in that case.

the method is getting a bit complex anyway, plus we have a branche that always return but still gets an else...
I'd suggest starting with a `interactionRegion->type == InteractionRegion::Type::Occlusion` condition, where we handle the simpler occlusion case and early return.
and then we can have all the consolidation/discovered region/guard code (everything that applies to `Type::Interaction`).